File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        int maxValue = 0;
        for (int i : tangerine) {
            if (maxValue < i) {
                maxValue = i;
            }
        }
        int[] arr = new int[maxValue + 1];
        for (int i = 0; i < tangerine.length; i++) {
            arr[tangerine[i]]++;
        }
        Arrays.sort(arr);
        for (int i = arr.length - 1; i >= 0; i--) {
            if (k > 0) {
                k -= arr[i];
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    max_idx = max(tangerine)
    cnt_arr = [0] * (max_idx+1)

    for i in range(len(tangerine)):
        cnt_arr[tangerine[i]] += 1

    cnt_arr.sort(reverse=True)
    for i in cnt_arr:
        if k > 0:
            k -= i
            answer += 1
    print(answer)
    return answer
